gene_finder.py

```diff
@@ -19,6 +19,7 @@ def shuffle_string(s):
 
```

Comment:
Overall - good implementation of dictionaries and list comprehensions, and general cleaning up.